main.cpp

```diff
@@ -88,6 +89,8 @@ struct Point
     float x{0}, y{0};
 };
 
+using FloatFunctionPtr = void(*)(float&);
+
```

Comment:
This should be scoped to inside the FloatType class.  it should not be a global alias.
- [ ] move alias.

Reply:
I use the same alias in the out of class implementation of `FloatType& FloatType::apply(FloatFunctionPtr);` on line 136. Should I move the function implementation inside the class or not use the alias in the out of class argument list?